pkg/controller/job/job_controller_test.go

```diff
@@ -109,14 +111,46 @@ func newJob(parallelism, completions, backoffLimit int32, completionMode batch.C
 	return newJobWithName("foobar", parallelism, completions, backoffLimit, completionMode)
 }
 
+type rateLimitingQueue struct {
```

Comment:
Instead of copying this implementation, it's better to add a function to provide a queue in `k8s.io/client-go/util/workqueue`. Then `NewController` can directly pass the clock interface too.

Reply:
```golang
func NewNamedRateLimitingQueueWithCustomClock(clock clock.WithTicker, rateLimiter RateLimiter, name string) RateLimitingInterface {
	return &rateLimitingType{
		DelayingInterface: NewDelayingQueueWithCustomClock(clock, name),
		rateLimiter:       rateLimiter,
	}
}
```
Will add something like so?

Reply:
Done